src/fluree/db/connection.cljc

```diff
@@ -111,7 +112,14 @@
                        tx-private-key tx-key-id
                        meta
                        add-listener remove-listener
-                       close])
+                       close]
+  #?@(:clj
+      [full-text/IndexConnection
+       (open-storage [conn network dbid lang]
+                     (-> conn
+                         :meta
+                         :file-storage-path
+                         (full-text/disk-index network dbid lang)))]))
```

Comment:
Or maybe check that we're using FS or mem storage? Or maybe it's time for a `:supports-full-text-index` bool (or similar) on the storage subsystems?

Reply:
Yeah, that's a good point. I think we could check explicitly for mem and fs and start the correct indexer accordingly. 

Ideally we'd have different connection records based on how they're backed, like a `MemoryConnection`, etc, and then implement the `full-text/IndexConnection` protocol individually and only on the ones that support it, but I think that's a bigger refactor outside of the scope of this bug. 

But checking for mem and fs, and throwing an exception otherwise is probably fine for now.